frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,106 @@
-          const sentDate = new Date().toString()
+        this.form.errorMsg = null
+        const groupId = this.currentGroupId
+        const groupName = this.currentGroupState.groupName
+
+        for (let member of this.form.invitees) {
+          let mailbox = member.state.attributes.mailbox
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('proposals/invite/mailRecord/create/v1', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('proposals/invite/groupRecord/create/v1', inviteToMailbox.toHash(), memberName, groupId)
+
+          if (this.isProposal) {
+            let latest = await backend.latestHash(groupId)
+            let proposal = new Events.HashableGroupProposal({
+              proposal: template(
+                L('This is a Vote for {name} to become a member of {group}'),
+                { name: memberName, group: groupName }
+              ),
+              // calculate the voting threshold from the group data
+              percentage: this.currentGroupState.memberApprovalPercentage * 0.01,
+              candidate: memberName,
+              actions: [
+               { contractId: mailbox, action: JSON.stringify(inviteToMailbox.toObject()) },
+               { contractId: groupId, action: JSON.stringify(inviteToGroup.toObject()) }
+              ],
+              initiator: this.loggedIn.name,
+              initiationDate: new Date().toString()
+            }, latest)
+            await backend.publishLogEntry(groupId, proposal)
+          } else {
+            await backend.publishLogEntry(mailbox, inviteToMailbox)
+            await backend.publishLogEntry(groupId, inviteToGroup)
```

Comment:
Recommend:

1. Switching the order of these two `publishLogEntries`, because it's easier to handle errors if the user doesn't get the invite, than if the user did but the group didn't get notified
2. Adding a comment `// TODO: place these in SBP calls for the ArchivedTransactionQueue`